packages/block-library/src/paragraph/test/edit.native.js

```diff
@@ -685,4 +685,38 @@ describe( 'Paragraph block', () => {
 		<!-- /wp:paragraph -->"
 	` );
 	} );
+
+	it( 'should focus on the previous Paragraph block when backspacing in an empty Paragraph block', async () => {
+		// Arrange
+		const screen = await initializeEditor();
+		await addBlock( screen, 'Paragraph' );
+
+		// Act
+		const paragraphBlock = getBlock( screen, 'Paragraph' );
+		fireEvent.press( paragraphBlock );
+		const paragraphTextInput =
+			within( paragraphBlock ).getByPlaceholderText( 'Start writing…' );
+		typeInRichText( paragraphTextInput, 'A quick brown fox jumps' );
+
+		await addBlock( screen, 'Paragraph' );
+		const secondParagraphBlock = getBlock( screen, 'Paragraph', {
+			rowIndex: 2,
+		} );
+		fireEvent.press( secondParagraphBlock );
+
+		const secondParagraphTextInput =
+			within( secondParagraphBlock ).getByPlaceholderText(
+				'Start writing…'
+			);
+		fireEvent( secondParagraphTextInput, 'onKeyDown', {
+			nativeEvent: {},
+			preventDefault() {},
+			keyCode: BACKSPACE,
+		} );
+
+		// Assert
+		// Check for formatting options to make sure the RichText Input is focused
+		const boldFormattingButton = screen.getByLabelText( 'Bold' );
+		expect( boldFormattingButton ).toBeVisible();
```

Comment:
I wonder if there would be other options to check if the text input is focused 🤔. Maybe spying a call to `TextInputState.focusTextInput`.

Reply:
Thanks for the suggestions! Updated in https://github.com/WordPress/gutenberg/pull/56496/commits/d01f0bdb27a4d0d08d6341b6a96d0709bb10bdeb